packages/mobile/src/send/SendAmount.tsx

```diff
@@ -157,24 +138,23 @@ export class SendAmount extends React.Component<Props, State> {
   state: State = {
     amount: '',
     reason: '',
-    characterLimitExceeded: false,
   }
 
   componentDidMount() {
     this.props.fetchDollarBalance()
-    this.fetchLatestPhoneAddress()
+    this.fetchLatestPhoneAddressesAndRecipientVerificationStatus()
   }
 
-  fetchLatestPhoneAddress = () => {
-    const recipient = this.getRecipient()
-    if (recipient.kind === RecipientKind.QrCode || recipient.kind === RecipientKind.Address) {
-      // Skip for QR codes or Addresses
-      return
-    }
-    if (!recipient.e164PhoneNumber) {
-      throw new Error('Missing recipient e164Number')
+  fetchLatestPhoneAddressesAndRecipientVerificationStatus = () => {
+    const { recipient } = this.props
+    // Skip phone number fetch for QR codes or Addresses
+    if (recipient.kind !== RecipientKind.QrCode && recipient.kind !== RecipientKind.Address) {
```

Comment:
Instead of the ! could you list which RecipientKinds would need a phone number?

Reply:
Good suggestion, much less confusing that way